lib/matplotlib/backend_bases.py

```diff
@@ -2948,12 +2950,15 @@ def _update_view(self):
             return
         # Retrieve all items at once to avoid any risk of GC deleting an Axes
         # while in the middle of the loop below.
+        subplotpars = {}
         items = list(nav_info.items())
-        for ax, (view, (pos_orig, pos_active)) in items:
+        for ax, (view, (pos_orig, pos_active), subplotpars) in items:
             ax._set_view(view)
             # Restore both the original and modified positions
             ax._set_position(pos_orig, 'original')
             ax._set_position(pos_active, 'active')
+
+        self.canvas.figure.subplotpars.update(**subplotpars)
```

Comment:
I'm a little confused why you don't just do all this with the `subplotpars` object directly, rather than making a dictionary and then passing one back in.  Does that not work for some reason?

Reply:
I am not sure if I thought about it but I have tested it now and it didn't work well in same cases for same reason. 

I also think that it feels better to read and copy the parameters and update the same object than copy the whole object and change object in `_update_view`. I am not sure now but it is possibly that gridspec has a view to the SubplotParams object in figure.

Reply:
This does seem very odd...

Reply:
This is very likely because subplotpars are mutable, so even if you push them onto the stack they'll get mutated after the fact unless you make a copy first.